bats/tests/helpers/vm.bash

```diff
@@ -10,18 +10,18 @@ wait_for_shell() {
 }
 
 factory_reset() {
-    rdctl factory-reset
-
-    if is_unix; then
-        if pgrep -f rancher-desktop; then
-            pkill -f rancher-desktop
-            # Re-run because RD was not stopped before deleting the data
-            rdctl factory-reset
+    if using_npm_run_dev; then
+        if is_unix; then
+            # Is there a better way to force-kill an `npm run dev` instance
+            # including Lima subprocesses?
+            run pkill -f "$(readlink -f "$PATH_RESOURCES")"
```

Comment:
I don't think this will catch what you want; the current list of things we have includes:
```
node node_modules/nuxt/bin/nuxt.js dev --hostname localhost --port 8888 …/rd/pkg/rancher-desktop
node --max-old-space-size=2048 …/rd/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporterRpcService.js
node node_modules/electron/cli.js …/rd 8888 …/rd/scripts/ts-wrapper.js …/rd/scripts/dev.ts
…/rd/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/MacOS/Electron Helper (GPU) --type=gpu-process <stuff elided>
…/rd/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS/Electron Helper --type=utility <stuff elided>
…/rd/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Helper (Renderer).app/Contents/MacOS/Electron Helper (Renderer) --type=renderer <stuff elided>
…/rd/resources/darwin/lima/bin/limactl --debug hostagent <stuff elided>
…/rd/resources/darwin/lima/bin/qemu-system-aarch64 <stuff elided>
```

This means it'll catch lima/qemu, but not the Electron bits, the nuxt dev server, etc.

Reply:
Did you actually try it? I tested it with this change:

```console
$ g diff
diff --git bats/tests/helpers/info.bash bats/tests/helpers/info.bash
index 3d65093f..44f2bbfc 100644
--- bats/tests/helpers/info.bash
+++ bats/tests/helpers/info.bash
@@ -17,3 +17,16 @@ load load.bash
 info() { # @test
     echo "Using '$RD_LOCATION' install; resources '$PATH_RESOURCES'" >&3
 }
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start Rancher Desktop' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+@test 'factory reset again' {
+    factory_reset
+}
$ RD_LOCATION=npm bats-core/bin/bats tests/helpers/info.bash
info.bash
 ✓ info
Using 'npm' install; resources '/Users/jan/suse/rd/resources'
 ✓ factory reset
 ✓ start Rancher Desktop
 ✓ factory reset again

4 tests, 0 failures
```

After the test finished, no Rancher Desktop processes were running.

Maybe there is a more elegant way to do this (see also comment inside the code), but it is outside the scope of this PR, which is fixing BATS for the non-npm use case.

The npm use case isn't implemented at all for Windows, and still experimental on Unix. I'll create separate issues to implement this.

Reply:
> After the test finished, no Rancher Desktop processes were running.

In a different test I saw some node processes still running, so I added a second `pkill`:

```
run pkill -f "$(readlink -f "$PATH_REPO_ROOT/node_modules")"
```